### PR TITLE
Interpretation of dynamic FieldControl annotations

### PR DESCRIPTION
**Pull request by HeikoTheissen**

### CODE REVIEW

File: vocabularies/Common.xml [hunk 1]
```diff
@@ -371,22 +371,33 @@
...
+          <Annotation Term="Core.LongDescription">
+            <String>A requested change of the property is rejected if this annotation is `ReadOnly` in the before-image _and_ the after-image of the request.
+            To statically mark a property as read-only use term [Core.Computed](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Core.V1.md#Computed) instead.</String>
+          </Annotation>
         </Member>
         <Member Name="Inapplicable" Value="0">
           <Annotation Term="Core.Description" String="Property has no meaning in the current entity state" />
           <Annotation Term="Core.LongDescription">
-            <String>This value does not make sense as a static annotation value.
+            <String>A requested change mentioning the property is rejected if this annotation is `Inapplicable` in the after-image of the request.
```
Comment (ralfhandl) [inline vocabularies/Common.xml:399]: What if the property value is set to `null` at the same time as setting the property to `Inapplicable`? That seems to make sense.

File: vocabularies/Common.xml [hunk 2]
```diff
@@ -371,22 +371,33 @@
...
         <Annotation Term="Core.Description" String="Control state of a property" />
+        <Annotation Term="Core.LongDescription">
+          <String>If this annotation is given dynamically in the metadata, its value in the before-image and after-image of a request
+          can play a role when changes to a property are requested.</String>
+        </Annotation>
         <Member Name="Mandatory" Value="7">
           <Annotation Term="Core.Description" String="Property is mandatory from a business perspective" />
-          <Annotation Term="Core.LongDescription" String="This annotation value does not imply any restrictions on the value range of the property. For restricting the value range use e.g. the standard type facet `Nullable` with a value of `false` to exclude the `null` value, or terms from the [Validation vocabulary](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Validation.V1.md)." />
+          <Annotation Term="Core.LongDescription">
+            <String>A requested removal of the property is rejected if this annotation is `Mandatory` in the after-image of the request.
```
Comment (ralfhandl) [inline vocabularies/Common.xml:381]: "removal" is a bit fuzzy, but so is "mandatory".

**Review by ralfhandl — DISMISSED** [review]

File: vocabularies/Common.xml [hunk 3]
```diff
@@ -371,22 +371,33 @@
...
         </Member>
         <Member Name="Optional" Value="3">
           <Annotation Term="Core.Description" String="Property may have a value" />
           <Annotation Term="Core.LongDescription" String="This value does not make sense as a static annotation value." />
         </Member>
         <Member Name="ReadOnly" Value="1">
           <Annotation Term="Core.Description" String="Property value cannot be changed" />
-          <Annotation Term="Core.LongDescription" String="To statically mark a property as read-only use term [Core.Computed](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Core.V1.md#Computed) instead" />
+          <Annotation Term="Core.LongDescription">
+            <String>A requested change of the property is ignored if this annotation is `ReadOnly` in the before-image _and_ the after-image of the request.
```
[the remark this answers is not in the document]
Comment (HeikoTheissen) [reply vocabularies/Common.xml:392]: That would be the "either" variant: If an after-image can be constructed where the property is not read-only or has not changed compared to the before-image, this is allowed even if the property was read-only in the before-image.

But it is harder to implement than the "or" variant.

File: vocabularies/Common.json [hunk 4]
```diff
@@ -304,20 +304,21 @@
...
       "Mandatory": 7,
       "Mandatory@Core.Description": "Property is mandatory from a business perspective",
-      "Mandatory@Core.LongDescription": "This annotation value does not imply any restrictions on the value range of the property. For restricting the value range use e.g. the standard type facet `Nullable` with a value of `false` to exclude the `null` value, or terms from the [Validation vocabulary](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Validation.V1.md).",
+      "Mandatory@Core.LongDescription": "A request that sets the property to its initial value or null fails entirely if this annotation is `Mandatory` in the after-image of the request.\n\n            This annotation value does not imply any restrictions on the value range of the property. For restricting the value range use e.g. the standard type facet `Nullable` with a value of `false` to exclude the `null` value, or terms from the [Validation vocabulary](https://github.com/oasis-tcs/odata-vocabularies/blob/main/vocabularies/Org.OData.Validation.V1.md).",
```
[the remark this answers is not in the document]
Comment (uhlmannm) [reply vocabularies/Common.json:310]: We discussed this. Whether a property is mandatory or inapplicable is (hopefully) controlled by the value of other properties. Hence in draft applications, there will be no issue as the client gets the update of the Field Control property before the user starts to work on the actual property. With non-draft applications, the whole concept of dynamic field control calculated in the backend is insufficient. An application needs to ensure that dynamic field control properties can be updated on the client when relevant other properties changed even though there is no PATCH.

**Review by uhlmannm — APPROVED** [review]

**Pull request merged**